### PR TITLE
[#2704] Filter contactmomenten by kanaal

### PR DESCRIPTION
**Pull request by pi-sigma**
Taiga: https://taiga.maykinmedia.nl/project/open-inwoner/task/2704

### CODE REVIEW

File: src/open_inwoner/openklant/clients.py [hunk 1]
```diff
@@ -159,6 +160,10 @@ def partial_update_klant(self, klant: Klant, update_data) -> Klant | None:
 
 
 class ContactmomentenClient(APIClient):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.klant_config = OpenKlantConfig.get_solo()
```
[the remark this answers is not in the document]
Comment (pi-sigma) [reply src/open_inwoner/openklant/clients.py]: D'accord with the second point, I was going back and forth between putting the functionality in the client and putting it in the view. I've now implemented the filtering in our wrapper function, which keeps the client clean and makes it explicit in the view that we're retrieving a filtered result. Not sure about the first point, but it doesn't matter here (the config isn't needed for anything else in the client).

File: src/open_inwoner/openklant/wrap.py [hunk 2]
```diff
@@ -20,6 +22,7 @@ def fetch_klantcontactmomenten(
     user_bsn: str | None = None,
     user_kvk_or_rsin: str | None = None,
     vestigingsnummer: str | None = None,
+    filter: str | None = None,
```
[the remark this answers is not in the document]
Comment (pi-sigma) [reply src/open_inwoner/openklant/wrap.py:25]: Rigth, forgot to update the annotation after code change.

File: src/open_inwoner/openklant/constants.py [hunk 3]
```diff
@@ -17,3 +17,20 @@ def safe_label(cls, value, default=""):
             if default:
                 return default
             return str(value).replace("_", " ").title()
+
+
+class ContactmomentKanaalEsuite(models.TextChoices):
```
Comment (alextreme) [inline src/open_inwoner/openklant/constants.py:22]: How sure are we that this enumeration is the same across multiple eSuite instances? As with most things eSuite, this is likely something which is configurable...

Comment (pi-sigma) [reply src/open_inwoner/openklant/constants.py:22]: Discussed, we'll replace the whitelist containing hard-coded strings with a blacklist to be entered by the admin (cf. https://taiga.maykinmedia.nl/project/open-inwoner/task/2724)

**Pull request merged**